libnvme compilation errors

Seeing the following libnvme compilation error on a SLES15 SP4 MU host:

```
# ninja -C .build
ninja: Entering directory `.build'
[2/3] Compiling C++ object test/test-cpp.p/cpp.cc.o
FAILED: test/test-cpp.p/cpp.cc.o
c++ -Itest/test-cpp.p -Itest -I../test -I. -I.. -Iccan -I../ccan -Isrc -I../src -Iinternal -I../internal -I/usr/include/json-c -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -g -DCCAN_LIST_DEBUG=1 -DCCAN_STR_DEBUG=1 -MD -MQ test/test-cpp.p/cpp.cc.o -MF test/test-cpp.p/cpp.cc.o.d -o test/test-cpp.p/cpp.cc.o -c ../test/cpp.cc
In file included from ../src/nvme/linux.h:14:0,
                 from ../src/libnvme.h:18,
                 from ../test/cpp.cc:10:
../src/nvme/ioctl.h: In function ‘int nvme_ns_mgmt_create(int, nvme_id_ns*, __u32*, __u32, __u8, nvme_ns_mgmt_host_sw_specified*)’:
../src/nvme/ioctl.h:3062:2: sorry, unimplemented: non-trivial designated initializers not supported
  };
  ^

ninja: build stopped: subcommand failed.
```

```
# rpm -qa | grep gcc
gcc-7-3.9.1.x86_64
gcc-c++-7-3.9.1.x86_64
libgcc_s1-11.2.1+git610-150000.1.6.6.x86_64
libstdc++6-pp-gcc11-11.2.1+git610-150000.1.6.6.x86_64
libstdc++6-devel-gcc7-7.5.0+r278197-4.30.1.x86_64
gcc7-7.5.0+r278197-4.30.1.x86_64
gcc7-c++-7.5.0+r278197-4.30.1.x86_64
```

--- a/src/nvme/ioctl.h
+++ b/src/nvme/ioctl.h
@@ -3058,6 +3058,8 @@ static inline int nvme_ns_mgmt_create(int fd, struct nvme_id_ns *ns,
 		.nsid = NVME_NSID_NONE,
 		.sel = NVME_NS_MGMT_SEL_CREATE,
 		.csi = csi,
+		.rsvd1 = { 0, },
+		.rsvd2 = NULL,
 		.data = data,
 	};
 
@@ -3087,6 +3089,9 @@ static inline int nvme_ns_mgmt_delete(int fd, __u32 nsid)
 		.nsid = nsid,
 		.sel = NVME_NS_MGMT_SEL_DELETE,
 		.csi = 0,
+		.rsvd1 = { 0, },
+		.rsvd2 = NULL,
+		.data = NULL,
 	};
 
 	return nvme_ns_mgmt(&args);

